types/react/test/next.tsx

```diff
@@ -36,7 +36,7 @@ function useExperimentalHooks() {
     // $ExpectType boolean
     const deferredToggle = React.useDeferredValue(toggle);
 
-    const [func] = React.useState(() => () => 0);
+    const [func] = React.useState<() => number>(() => () => 0);
```

Comment:
Does this need the explicit type parameter now?

Reply:
Yes, omit the type parameter, then inferred as type `() => 0`.

This changes, but (IMO) it's better not to rely on this, and it passed its dependents' tests as well.